buildSrc/src/main/java/com/osacky/flank/gradle/YamlWriter.kt

```diff
@@ -27,167 +29,92 @@ internal class YamlWriter {
+    appendProperty(config.resultsBucket, name = "results-bucket")
+    appendMapProperty(config.environmentVariables, name = "environment-variables") {
+      appendln("    ${it.key}: ${it.value}")
+    }
+    appendListProperty(config.testTargets, name = "test-targets") { appendln("  - $it") }
+    appendListProperty(config.directoriesToPull, name = "directories-to-pull") { appendln("  - $it") }
+    appendProperty(config.flakyTestAttempts, name = "num-flaky-test-attempts")
+    appendProperty(config.resultsDir, name = "results-dir")
+    appendProperty(config.testRunnerClass, name = "test-runner-class")
+    appendProperty(config.numUniformShards, name = "num-uniform-shards")
+    appendMapProperty(config.clientDetails, name = "client-details") { appendln("    ${it.key}: ${it.value}") }
+    appendMapProperty(config.otherFiles, name = "other-files") { appendln("    ${it.key}: ${it.value}") }
+    appendProperty(config.networkProfile, name = "network-profile")
+    appendProperty(config.roboScript, name = "robo-script")
+    appendListProperty(config.roboDirectives, name = "robo-directives") {
+      val value = it.getOrElse(2) { "" }.let { stringValue -> if (stringValue.isBlank()) "\"\"" else stringValue }
+      appendln("    ${it[0]}:${it[1]}: $value")
     }
+  }
 
-    config.networkProfile?.let {
-      appendln("  network-profile: $it")
-    }
+  private fun <T> StringBuilder.appendProperty(prop: Property<T>, name: String) {
+    if (prop.isPresent) appendln("  $name: ${prop.get()}")
+  }
 
-    config.roboScript?.let {
-      appendln("  robo-script: $it")
+  private fun <T, K> StringBuilder.appendMapProperty(
+    prop: MapProperty<T, K>,
+    name: String,
+    custom: StringBuilder.(Map.Entry<T, K>) -> Unit
+  ) {
+    if (prop.isPresentAndNotEmpty) {
+      appendln("  $name:")
+      prop.get().forEach { custom(it) }
     }
+  }
 
-    if (config.roboDirectives.isNotEmpty()) {
-      appendln("  robo-directives:")
-      config.roboDirectives.forEach {
-        val value = it.getOrElse(2) { "" }.let { stringValue -> if (stringValue.isBlank()) "\"\"" else stringValue }
-        appendln("    ${it[0]}:${it[1]}: $value")
-      }
+  private fun <T> StringBuilder.appendListProperty(
+    prop: ListProperty<T>,
+    name: String,
+    custom: StringBuilder.(T) -> Unit
+  ) {
+    if (prop.isPresentAndNotEmpty) {
+      appendln("  $name:")
+      prop.get().forEach { custom(it) }
     }
   }
 
-  private fun flakyTestAttemptsLine(flakyTestAttempts: Int): String {
-    val label = "num-flaky-test-attempts"
-    return "  $label: $flakyTestAttempts"
-  }
+  private val <T> ListProperty<T>.isPresentAndNotEmpty
+    get() = isPresent && get().isEmpty().not()
 
-  private fun repeatTestsLine(repeatTests: Int): String {
-    val label = "num-test-runs"
-    return "  $label: $repeatTests"
-  }
+  private val <T, K> MapProperty<T, K>.isPresentAndNotEmpty
+    get() = isPresent && get().isEmpty().not()
```

Comment:
same suggestion here:
```suggestion
    get() = isPresent && get().isNotEmpty()
```